src/main/java/org/tugraz/sysds/runtime/lineage/LineageCacheConfig.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2019 Graz University of Technology
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.tugraz.sysds.runtime.lineage;
+
+import org.tugraz.sysds.api.DMLScript;
+import java.util.ArrayList;
+
+public class LineageCacheConfig {
+	public enum cacheType {
+		FULL,   // no rewrites
+		PARTIAL, 
+		BOTH,
+		NONE
+	}
+	
+	public ArrayList<String> _MMult = new ArrayList<String>();
+	
+	public enum cachedItemsL1 {
```

Comment:
Please follow naming conventions - start Enums with a capital letter. Also, maybe it would be good to call it head and tail?